src/compiler/watch.ts

```diff
@@ -708,7 +758,7 @@ namespace ts {
 
             // Create new source file if requested or the versions dont match
             if (!hostSourceFile || shouldCreateNewSourceFile || !isFilePresentOnHost(hostSourceFile) || hostSourceFile.version.toString() !== hostSourceFile.sourceFile.version) {
-                const sourceFile = getNewSourceFile();
+                const sourceFile = getNewSourceFile.call(compilerHost, fileName, languageVersion, onError);

```

Comment:
The host we create from `createCompilerHostFromProgramHost` doesn't use `this` (all of its methods are arrows). Do we need the indirect call via `Function#call` here?